sdk/identity/azure-identity/tests/test_token_credentials_env.py

```diff
@@ -116,3 +116,158 @@ def test_token_credentials_env_with_exclude():
         actual_classes = {c.__class__ for c in credential.credentials}
 
         assert EnvironmentCredential not in actual_classes
+
+
+def test_token_credentials_env_workload_identity_credential():
+    """With AZURE_TOKEN_CREDENTIALS=WorkloadIdentityCredential, only WorkloadIdentityCredential should be used"""
+
+    with patch.dict(
+        "os.environ",
+        {
+            EnvironmentVariables.AZURE_TOKEN_CREDENTIALS: "WorkloadIdentityCredential",
+            EnvironmentVariables.AZURE_AUTHORITY_HOST: "https://login.microsoftonline.com",
+            EnvironmentVariables.AZURE_TENANT_ID: "tenant-id",
+            EnvironmentVariables.AZURE_CLIENT_ID: "client-id",
+            EnvironmentVariables.AZURE_FEDERATED_TOKEN_FILE: "/tmp/token",
+        },
+        clear=False,
+    ):
+        credential = DefaultAzureCredential()
+
+        # Get the actual credential classes in the chain
+        actual_classes = {c.__class__ for c in credential.credentials}
```

Comment:
Why a set instead of a list? We'd want to know if the chain had multiple instances of the same class

Reply:
Good point. Updated to use a list and adjusted some tests to check for duplication.